examples/gno.land/r/x/manfred_upgrade_patterns/upgrade_g/store/store.gno

```diff
@@ -0,0 +1,42 @@
+package store
+
+import (
+	"std"
+	"fmt"
+
+	"gno.land/r/x/manfred_upgrade_patterns/upgrade_g/admin"
+)
+
+var counterValue uint64
+
+var _ admin.Store = &store{}
+
+var currentStore *store
+
+type store struct {}
+
+func (s *store) GetCounter() uint64 {
+	if s != currentStore {
+		panic("Revoked store")
```

Comment:
When is this possible?

Reply:
The user tries to call the public function in a old version logics without going through the admin interface. In this case, any outer-world call would fail because the prevRealm will return the logic's address, not the admin's. However, the old logics still holds the store reference, thus need to be deprecated when a new store struct is instantiated.